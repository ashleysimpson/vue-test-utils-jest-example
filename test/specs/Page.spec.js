import {
  mount, createLocalVue, shallowMount
} from '@vue/test-utils'
import Vuex from 'vuex'
import Page from '@/components/Page.vue'
import {
  createRenderer
} from 'vue-server-renderer'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('List.vue', () => {
  it('matches snapshop shallow mount', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(Page, { localVue })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('matches snapshot mount', () => {
    const renderer = createRenderer()
    const wrapper = mount(Page, { localVue })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
