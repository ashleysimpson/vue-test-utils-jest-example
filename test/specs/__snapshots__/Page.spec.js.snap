// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshop shallow mount 1`] = `
<div>
    <message-toggle-stub></message-toggle-stub>
    <list-stub items="list item 1,listen item 2"></list-stub>
</div>
`;

exports[`List.vue matches snapshot mount 1`] = `
<div>
    <div>
        <h1>default message</h1> <button id="toggle-message">
            Change message
        </button>
    </div>
    <ul>
        <li>
            list item 1
        </li>
        <li>
            listen item 2
        </li>
    </ul>
</div>
`;
